dayIcons.js

```diff
@@ -37,30 +36,39 @@ const mostFrequentCode = (groupedWeatherCodes) => {
 
 const getWeatherDayIcon = (hours) => {
 	let mainHours = getMainHours(hours);
+	let dayOrNightIcon = mainHours.length === 0 ? iconCode.night : iconCode.day;
 	
 	if(mainHours.length === 0) {
 		mainHours = hours;
-		return iconCode.night[mostFrequentCode(groupWeatherCodes(mainHours))]
 	};
-	
+	const weatherCode = mostFrequentCode(groupWeatherCodes(mainHours));
+	const weatherIcon= dayOrNightIcon[weatherCode];
```

Comment:
You unnecessarily do it here - move those two lines down after all the checks, near the end of the function.
Ex. if you have rain, you unnecessarily calculate  average weather icon here!

Reply:
Good point. Changed.

Reply:
Good point, done.